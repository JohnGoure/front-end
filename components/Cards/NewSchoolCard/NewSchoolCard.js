import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Button from 'common/components/Button/Button';
import Card from 'common/components/Card/Card';
import OutboundLink from 'common/components/OutboundLink/OutboundLink';
// import DesktopIcon from 'static/images/icons/FontAwesome/desktop-solid.svg';
// import DevicesIcon from 'static/images/icons/FontAwesome/devices-alt-solid.svg';
// import HomeIcon from 'static/images/icons/FontAwesome/home-solid.svg';
// import PeopleIcon from 'static/images/icons/FontAwesome/users-solid.svg';
// import GIBillApprovedIcon from 'static/images/icons/gi-bill-approved.svg';
// import GIBillUnavailableIcon from 'static/images/icons/gi-bill-unavailable.svg';
import styles from './NewSchoolCard.css';

class NewSchoolCard extends Component {
  static propTypes = {
    hasHardwareIncluded: PropTypes.bool.isRequired,
    hasOnline: PropTypes.bool.isRequired,
    isFullTime: PropTypes.bool.isRequired,
    locations: PropTypes.arrayOf(
      PropTypes.shape({
        address: PropTypes.string,
        city: PropTypes.string,
        doesAcceptGIBill: PropTypes.bool,
        state: PropTypes.string,
      }),
    ),
    logoSource: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    website: PropTypes.string.isRequired,
  };

  static defaultProps = {
    locations: [],
  };

  state = {
    isCardFlipped: false,
  };

  showLocations = () => {
    this.setState({ isCardFlipped: true });
  };

  render() {
    const { props, state } = this;

    if (state.isCardFlipped) {
      return <Card className={styles.NewSchoolCard} hasAnimationOnHover />;
    }

    return (
      <Card className={styles.NewSchoolCard} hasAnimationOnHover>
        <img src={props.logoSource} alt={`${props.name} Logo`} className={styles.logo} />

        <div className={styles.flexContainer}>
          <Button>
            <OutboundLink
              analyticsEventLabel={`Code School Website: ${props.name}`}
              hasIcon={false}
              href={props.website}
            >
              Go To Website
            </OutboundLink>
          </Button>

          {props.locations.length > 1 ? (
            <Button
              analyticsObject={{
                action: 'Button Selected',
                category: 'Interactions',
                label: `Code School Locations: ${props.name}`,
              }}
              onClick={this.showLocations}
            >
              See Locations
            </Button>
          ) : (
            <span className={styles.singleLocationText}>
              {props.locations.length === 1
                ? `${props.locations[0].city}, ${props.locations[0].state}`
                : 'Online Only'}
            </span>
          )}
        </div>

        <hr className={styles.divider} />

        <section className={classNames(styles.flexContainer, styles.detailIcons)}>
          <div />
        </section>

        {/* <template className={styles.content}>

          <section className={styles.NewSchoolCardText}>
            <h5 className={styles.name}>{name}</h5>
            <address className={styles.location}>
              {hasOnlineProgram && 'Online Available'}
              {hasOnlineProgram && <br />}
              {city && `${city}, `}
              {state}
            </address>

            <ul className={styles.info}>
              <li>
                GI Bill Accepted: <b>{doesAcceptGIBill ? 'Yes' : 'No'}</b>
              </li>
              <li>
                Commitment: <b>{isFullTime ? 'Full-Time' : 'Flexible'}</b>
              </li>
              <li>
                Hardware Included: <b>{hasHardwareIncluded ? 'Yes' : 'No'}</b>
              </li>
            </ul>
          </section>
        </template> */}
      </Card>
    );
  }
}

export default NewSchoolCard;
