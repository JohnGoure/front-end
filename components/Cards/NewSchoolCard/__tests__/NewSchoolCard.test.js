/* eslint-env jest */
import React from 'react';
import s3 from 'common/constants/urls';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import NewSchoolCard from '../NewSchoolCard';

describe('NewSchoolCard', () => {
  test('should render properly with all props assigned', () => {
    createSnapshotTest(
      <NewSchoolCard
        website="https://www.codeplatoon.org"
        name="Code Platoon"
        address="123 Baker Street"
        schoolCity="Chicago"
        schoolState="IL"
        logoSource={`${s3}codeSchoolLogos/code_platoon.jpg`}
        doesAcceptGIBill
        isFullTime
        hasHardware={false}
      />,
    );
  });

  test('should render properly with all required assigned', () => {
    createSnapshotTest(
      <NewSchoolCard
        website="https://www.codeplatoon.org/not-real"
        name="Code Platoon Fake"
        address="123 Baker Street"
        logoSource={`${s3}codeSchoolLogos/code_platoon.jpg`}
        doesAcceptGIBill={false}
        isFullTime={false}
        hasHardware
      />,
    );
  });
});
