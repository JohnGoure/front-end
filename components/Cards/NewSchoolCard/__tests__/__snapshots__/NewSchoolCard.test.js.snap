// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewSchoolCard should render properly with all props assigned 1`] = `
<article
  className="Card NewSchoolCard animatedCard"
>
  <img
    alt="Code Platoon Logo"
    className="logo"
    src="undefinedcodeSchoolLogos/code_platoon.jpg"
  />
  <div
    className="flexContainer"
  >
    <button
      className="Button primary"
      disabled={false}
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      <a
        className=""
        href="https://www.codeplatoon.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          className="screenReaderOnly"
        >
          Opens in new window
        </span>
        Go To Website
      </a>
    </button>
    <span
      className="singleLocationText"
    >
      Online Only
    </span>
  </div>
  <hr
    className="divider"
  />
  <section
    className="flexContainer detailIcons"
  >
    <div />
  </section>
</article>
`;

exports[`NewSchoolCard should render properly with all required assigned 1`] = `
<article
  className="Card NewSchoolCard animatedCard"
>
  <img
    alt="Code Platoon Fake Logo"
    className="logo"
    src="undefinedcodeSchoolLogos/code_platoon.jpg"
  />
  <div
    className="flexContainer"
  >
    <button
      className="Button primary"
      disabled={false}
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      <a
        className=""
        href="https://www.codeplatoon.org/not-real"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          className="screenReaderOnly"
        >
          Opens in new window
        </span>
        Go To Website
      </a>
    </button>
    <span
      className="singleLocationText"
    >
      Online Only
    </span>
  </div>
  <hr
    className="divider"
  />
  <section
    className="flexContainer detailIcons"
  >
    <div />
  </section>
</article>
`;
